src/commons/build-partial-update-state-command.ts

```diff
@@ -0,0 +1,59 @@
+import { StateDetails } from '../data-types/interfaces';
+import { UpdateItemCommandInput } from '@aws-sdk/client-dynamodb';
+import { getCurrentTimestamp } from './get-current-timestamp';
+import { EventsEnum, MetricNames } from '../data-types/constants';
+import { logAndPublishMetric } from './metrics';
+
+/**
+ * Method to build a Partial of UpdateItemCommandInput
+ * @param newState - new account state object
+ * @param eventName - the name of the event received
+ * @param interventionName - optional intervention name if the event was a fraud intervention
+ */
+export const buildPartialUpdateAccountStateCommand = (
+  newState: StateDetails,
+  eventName: EventsEnum,
+  interventionName?: string,
+): Partial<UpdateItemCommandInput> => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const base: Record<string, any> = {
+    ExpressionAttributeNames: {
+      '#B': 'blocked',
+      '#S': 'suspended',
+      '#RP': 'resetPassword',
+      '#RI': 'reproveIdentity',
+      '#UA': 'updatedAt',
+    },
+    ExpressionAttributeValues: {
+      ':b': { BOOL: newState.blocked },
+      ':s': { BOOL: newState.suspended },
+      ':rp': { BOOL: newState.resetPassword },
+      ':ri': { BOOL: newState.reproveIdentity },
+      ':ua': { N: `${getCurrentTimestamp().milliseconds}` },
+    },
+    UpdateExpression: 'SET #B = :b, #S = :s, #RP = :rp, #RI = :ri, #UA = :ua',
+  };
+  if (eventName === EventsEnum.IPV_IDENTITY_ISSUED) {
+    base['ExpressionAttributeNames']['#RIdA'] = 'reprovedIdentityAt';
+    base['ExpressionAttributeValues'][':rida'] = { N: `${getCurrentTimestamp().milliseconds}` };
+    base['UpdateExpression'] += ', #RIdA = :rida';
+  } else if (eventName === EventsEnum.AUTH_PASSWORD_RESET_SUCCESSFUL) {
+    base['ExpressionAttributeNames']['#RPswdA'] = 'resetPasswordAt';
+    base['ExpressionAttributeValues'][':rpswda'] = { N: `${getCurrentTimestamp().milliseconds}` };
+    base['UpdateExpression'] += ', #RPswdA = :rpswda';
+  } else {
+    if (!interventionName) {
+      logAndPublishMetric(MetricNames.INTERVENTION_DID_NOT_HAVE_NAME_IN_CURRENT_CONFIG);
+      throw new Error('intervention received did not have an interventionName field');
```

Comment:
Could we update this error line to include capitals and full stops, so something like:
"The received intervention did not contain a field for interventionName."